docs/source/starter/style_guide.rst

```diff
@@ -173,55 +151,57 @@ In practice, this code looks like:
+We recommend using forward for inference/predictions and keeping training_step independent
 
 .. code-block:: python
 
     def forward(self, x):
         embeddings = self.encoder(x)
-        return embeddings
 
 
-    def training_step(self, batch, batch_idx):
-        x, y = batch
+    def training_step(self):
+        x, y = ...
         z = self.encoder(x)
         pred = self.decoder(z)
         ...
 
+However, when using DataParallel, you will need to call forward manually
+
+.. code-block:: python
+
+    def training_step(self):
+        x, y = ...
+        z = self(x)  # < ---------- instead of self.encoder(x)
+        pred = self.decoder(z)
+        ...
 
 --------------
 
 ****
 Data
 ****
-
 These are best practices for handling data.
 
 Dataloaders
 ===========
-
-Lightning uses :class:`~torch.utils.data.DataLoader` to handle all the data flow through the system. Whenever you structure dataloaders,
+Lightning uses dataloaders to handle all the data flow through the system. Whenever you structure dataloaders,
 make sure to tune the number of workers for maximum efficiency.
 
-.. warning:: Make sure not to use ``Trainer(strategy="ddp_spawn")`` with ``num_workers>0`` in a DataLoader or you will bottleneck your code.
+.. warning:: Make sure not to use ddp_spawn with num_workers > 0 or you will bottleneck your code.
 
 DataModules
 ===========
+Lightning introduced datamodules. The problem with dataloaders is that sharing full datasets is often still challenging
+because all these questions need to be answered:
 
-The :class:`~pytorch_lightning.core.datamodule.LightningDataModule` is designed as a way of decoupling data-related
-hooks from the :class:`~pytorch_lightning.core.lightning.LightningModule` so you can develop dataset agnostic models. It makes it easy to hot swap different
-datasets with your model, so you can test it and benchmark it across domains. It also makes sharing and reusing the exact data splits and transforms across projects possible.
-
-Check out :ref:`data` document to understand data management within Lightning and its best practices.
-
-------------
+* What splits were used?
+* How many samples does this dataset have?
+* What transforms were used?
 
-********
-Examples
-********
+It's for this reason that we recommend you use datamodules. This is especially important when collaborating because
+it will save your team a lot of time as well.
 
-Checkout the live examples to get your hands dirty:
+All they need to do is drop a datamodule into a lightning trainer and not worry about what was done to the data.
 
-- `Introduction to PyTorch Lightning <https://pytorch-lightning.readthedocs.io/en/stable/notebooks/lightning_examples/mnist-hello-world.html>`_
-- `Introduction to DataModules <https://pytorch-lightning.readthedocs.io/en/stable/notebooks/lightning_examples/datamodules.html>`_
+This is true for both academic and corporate settings where data cleaning and ad-hoc instructions slow down the progress
```

Comment:
This section is a good start but I would add more content to it. Also, let's keep hyperlinks.